### ISSUE TITLE
Question about the evaluation of pose, the optimized scale

### ISSUE THREAD

**Issue by kwea123** [nianticlabs/monodepth2#44]
In https://github.com/nianticlabs/monodepth2/blob/1cc8b81b6ca46fc6878de92ff126c7d2d94f5f89/evaluate_pose.py#L43
The scale is optimized for every 5-frame `track_length`. I wonder if this is a common way to do (like the code is borrowed from SfMLearner); in my opinion it is sort of "cheating" (no offense), because in reality you cannot optimize the scale all the time since no ground truth is available. A more reasonable way should be to fix the scale beforehand, e.g. using the optimal scale on the training set, and use the same scale for all the sequence.

What's your opinion?

**Comment by mdfirman** [comment]
The protocol we use for evaluating pose was established in _Unsupervised Learning of Depth and Ego-Motion from Video_, and re-used by almost all KITTI mono-trained papers since. We use this same evaluation method to enable a fair comparison with these papers. 

I would agree that being allowed to 'rescale' each 5-frame sequence gives some information away from the test set, but importantly: If the predicted poses are bad, no amount of rescaling is likely give you good scores. So I would not say this is cheating, but it isn't as thorough an evaluation as it could be with a more complex evaluation scenario such as the one you describe.

The issue with rescaling at test time also applies to the depth image predictions, where each depth image from mono-trained models are independently rescaled. We flagged this in our paper, and we introduce a novel evaluation where all test predictions are rescaled with a single rescaling factor (Section D.2).

**Comment by kwea123** [comment]
Thank you for referencing the original paper. I pose this question because without knowing that the scale is actually optimized every 5-frame and only seeing the good result, I have the impression that "Oh, so now it is better than slam methods?"

So here ORB-slam is without any ground truth optimized scale, and all other methods are with optimized scale right? I tried using a fixed scale optimized on the training set and evaluated Monodepth2 and the result is around 0.049+-0.035, which is not as good as ORB-slam. I have seen many (blog) posts saying that "monodepth is now better than slam in terms of visual odometry", so these statements are actually wrong, aren't they?

**Comment by daniyar-niantic** [comment]
Hi @kwea123
If you are interested in odometry estimated with neural networks, you should check out DVSO paper https://arxiv.org/abs/1807.02570.

**Comment by mdfirman** [comment]
Do not forget that ORB-SLAM applied to monocular sequences is also ambiguous up to scale, so would need to be rescaled somehow for evaluation. 

> I have seen many (blog) posts saying that "monodepth is now better than slam in terms of visual odometry"...

Do you think you could provide a link to this? 

Our paper is primarily about depth estimation. For understanding what is state of the art in pose estimation, I would recommend a paper which has a pose focus, such as the one Daniyar linked.
